app/javascript/utilities/mergeRefs.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Helper function to merge an array of refs when attaching to a Preact component. Useful for when a ref is used by both the component itself and its parent.
+ *
+ * @param {Array} refs Array of all references
+ *
+ * @example
+ * const MyComponent = forwardRef((props, forwardedRef) => {
+ *  const innerRef = useRef(null);
+ *  return (
+ *   <InnerComponent
+ *      ref={mergeRefs([forwardedRef, innerRef])}
+ *   />
+ *  );
+ * });
+ */
+export const mergeRefs = (refs) => (value) => {
```

Comment:
Question (non-blocking): From what I can tell this pattern has been introduced to keep the plain textarea and combobox input textarea's `value` property in sync?

Reply:
I introduced it so that in `EditorBody` we can continue to handle the drag and drop / image copy/paste - those hooks need a handle to the current textarea, which it has by passing a ref down to the MentionAutocomplete component (which then needs to be attached to the same textarea as the internal `plaintTextAreaRef`).

I think the alternative would be to handle that functionality like the auto-resize and make 'switching it on' a prop e.g. `allowPastedImages` / `allowDroppedImages` (or something), and use the relevant hooks directly in the MentionAutocompleteTextArea component, rather than in `EditorBody`. Feels like we'd be giving a lot of responsibility to the MentionAutocompleteTextAreaComponent if we do that though

Reply:
Refactor (blocking): `mergeRefs` is currently only used in the `<EditorBody />` component. Keep it as a function within that component for now as we may never use this elsewhere ([YAGNI](https://martinfowler.com/bliki/Yagni.html)). If we see this pattern pop up again, we can consider it as a utility function.